### ISSUE TITLE
Consider mentioning pip secure installs

### ISSUE THREAD

**Issue by matthewfeickert** [klieret/everything-you-didnt-now-you-needed#9]
In the slides you're mentioning lock files and `pip-tools` already, which is great!

https://github.com/klieret/everything-you-didnt-now-you-needed/blob/8ad55c208b5e370dfda676d7b2d58d9d556c3597/slides.md?plain=1#L389-L397

Though it might also be worth mentioning the benefits that arise from using ["secure installs" with `pip`](https://pip.pypa.io/en/stable/topics/secure-installs/).

c.f. also:
* https://github.com/brettcannon/pip-secure-install
* My SciPy 2022 "Analysis as Applications: Quick introduction to lockfiles" lightning talk that advocates for use of `pip`'s secure install recipes.

**Comment by klieret** [comment]
Thanks for the suggestion :)   So basically you would recommend that whenever a lock file is used, that all of the hashes are pinned as well? 

Generally though, I think this is out of scope for the next iteration of this material in October (which will be more hands-on and of smaller scope) but maybe a nice addition for the future!

**Comment by klieret** [comment]
Let me add the link to your slides here so I find them later: https://zenodo.org/record/6946666#.YvXRS9LMJhE

**Comment by matthewfeickert** [comment]
> So basically you would recommend that whenever a lock file is used, that all of the hashes are pinned as well?

Yes. If you don't have things pinned down to the hash level, then it isn't really a lock file.
